compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/repositories/RemotePackageRepository.java

```diff
@@ -242,27 +243,35 @@ private PackageNameResolutionRequest toPackageNameResolutionRequest(List<ImportM
         if (options.offline()) {
             return cachedPackages;
         }
-
-        // Resolve the requests from remote repository
-        try {
-            PackageResolutionRequest packageResolutionRequest = toPackageResolutionRequest(requests);
-            PackageResolutionResponse packageResolutionResponse = client.resolveDependencies(
-                    packageResolutionRequest, JvmTarget.JAVA_11.code(),
-                    RepoUtils.getBallerinaVersion(), true);
-
-            Collection<PackageMetadataResponse> remotePackages =
-                    fromPackageResolutionResponse(requests, packageResolutionResponse);
-            // Merge central requests and local requests
-            // Here we will pick the latest package from remote or local
-            return mergeResolution(remotePackages, cachedPackages);
-
-        } catch (ConnectionErrorException e) {
-            // ignore connect to remote repo failure
-            // TODO we need to add diagnostics for resolution errors
-        } catch (CentralClientException e) {
-            throw new ProjectException(e.getMessage());
+        // Remove the already resolved requests when the locking mode is hard
+        for (PackageMetadataResponse response : cachedPackages) {
+            if (response.packageLoadRequest().packageLockingMode().equals(PackageLockingMode.HARD)
+                    && response.resolutionStatus().equals(ResolutionResponse.ResolutionStatus.RESOLVED)) {
+                requests.remove(response.packageLoadRequest());
```

Comment:
A better approach would be to create a new list here. Modifying the method parameters should be avoided as much as possible.

Reply:
fixed with 68e2b272179cba27dae47408c84cb84d3e85adbc